.github/workflows/bump_torch_mlir.yml

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Automatically bumps the torch-mlir submodule to match shark-infra/torch-mlir:main.
+# Manually triggered for now, but can be scheduled in the future.
+# This typically takes ~90s to run.
+name: Bump submodule - shark-infra/torch-mlir
+
+on:
+  # TODO: Uncomment the following lines to enable the schedule when this action is stable & useful enough
+  # schedule:
+  #   - cron: '0 0 * * 0,4'  # Run at 00:00 on Sunday and Thursday
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - .github/workflows/bump_torch_mlir.yml
+
+jobs:
+  bump_submodule:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c  # v3.3.0
+        with:
+          submodules: true
+
+      - name: Setting git config
+        run: |
+          git config --local user.email "iree-github-actions-bot@google.com"
+          git config --local user.name "Submodule Bump Action"
+
+      - name: Bump torch-mlir submodule
+        id: bump_submodule
+        run: |
+          cd third_party/torch-mlir
+          git fetch origin
+          git checkout origin/main
+          TORCH_MLIR_COMMIT=$(git rev-parse HEAD)
```

Comment:
(can be a follow-up)

With workflow_dispatch, you could pass this in as an argument, defaulting to `HEAD`

https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#defining-inputs-for-manually-triggered-workflows